### PR TITLE
Create-mock-update-marks-endpoint: add update  marks endpoint

### PR DESCRIPTION
**Pull request by AnastasiyaPoleshuk**

### CODE REVIEW

File: updateMarks/updateMarks.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      SubjectName: constants.CONSTANTS.MOCK_SUBJECTS[id - 1].SubjectName,
+      AverageMark: averageMark,
+      Marks: marks
+    };
+    response.status = statusOk;
+  }
+  return response;
+};
+
+function calcAverageMark(marks) {
```
Comment (AyvengoBY) [inline updateMarks/updateMarks.js]: Its better to use **reduce** array extension :
**_const average = list => list.reduce((prev, curr) => prev + curr) / list.length;_**

File: updateMarks/updateMarks.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  if (id > 0 && id <= constants.CONSTANTS.MOCK_SUBJECTS.length) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function getMarksBySubjectId(id, marks, token) {
+  const statusOk = 200;
+  const statusUnauthorized = 401;
```
Comment (AyvengoBY) [inline updateMarks/updateMarks.js]: The same comment as in other  PRs

File: updateMarks/updateMarks.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  const response = {
```
Comment (AyvengoBY) [inline updateMarks/updateMarks.js:27]: Please process token first

File: updateMarks/updateMarks.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    const responseData = getMarksBySubjectId(subjectId, marks, token);
+    res.status(responseData.status).send(responseData.marksData);
+  }
+
+}
+
+function checkSubjectId(id) {
+  if (id > 0 && id <= constants.CONSTANTS.MOCK_SUBJECTS.length) {
+    return true;
+  } else {
```
Comment (AyvengoBY) [inline updateMarks/updateMarks.js]: No need to use 'else'

**Review by AyvengoBY — CHANGES_REQUESTED** [review]

File: updateMarks/updateMarks.js (hunk 2 above)
Comment (AnastasiyaPoleshuk) [reply updateMarks/updateMarks.js]: fixed

File: updateMarks/updateMarks.js (hunk 1 above)
Comment (AnastasiyaPoleshuk) [reply updateMarks/updateMarks.js]: changed

File: updateMarks/updateMarks.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  return true;
+}
+
+function getMarksBySubjectId(id, marks) {
+  const response = {
+    marksData: {},
+    status: StatusCodes.StatusCodes.UNAUTHORIZED,
+  };
+  if (id > 0 && id <= constants.CONSTANTS.MOCK_SUBJECTS.length) {
+    const digits = 1;
```
Comment (AyvengoBY) [inline updateMarks/updateMarks.js]: Could you please move this constant into constants.CONSTANTS. We need this approach for easy application setup in one place...

File: updateMarks/updateMarks.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+const updateMarks = (req, res) => {
+  const token = req.body.token;
+  const subjectId = req.body.subjectId;
+  const marks = req.body.marks;
+  const isRightToken = checkToken(token);
+
+  if (!isRightToken) {
+    const responseData = { message: "invalid token" };
+    res.status(404).send(responseData);
```
Comment (AyvengoBY) [inline updateMarks/updateMarks.js]: Please, use StatusCodes  here

**Review by AyvengoBY — CHANGES_REQUESTED** [review]

File: updateMarks/updateMarks.js (hunk 5 above)
Comment (AnastasiyaPoleshuk) [reply updateMarks/updateMarks.js]: fixed

File: updateMarks/updateMarks.js (hunk 6 above)
Comment (AnastasiyaPoleshuk) [reply updateMarks/updateMarks.js]: fixed

**Review by AyvengoBY — APPROVED** [review]

**Pull request merged**